app/code/community/Algolia/Algoliasearch/Helper/Algoliahelper.php

```diff
@@ -84,6 +84,11 @@ public function mergeSettings($index_name, $settings)
         } catch (\Exception $e) {
         }
 
+        if (isset($onlineSettings['searchableAttributes']) && isset($settings['attributesToIndex'])) {
+            $settings['searchableAttributes'] = $settings['attributesToIndex'];
+            unset($settings['attributesToIndex']);
+        }
+
```

Comment:
I would go even further than that and replace by searchAttributes even when not the both of them are set